src/Utilities/mutable_arithmetics.jl

```diff
@@ -67,13 +67,31 @@ end
 
 # To avoid type piracy, we add at least one `ScalarLike` outside of the `...`.
 const PROMOTE_IMPLEMENTED_OP = Union{typeof(+),typeof(-),typeof(*),typeof(/)}
+
+function MA.promote_operation(
+    op::PROMOTE_IMPLEMENTED_OP,
+    F::Type{MOI.ScalarAffineFunction{T}},
+    G::Type{MOI.VariableIndex},
+) where {T}
+    return promote_operation(op, T, F, G)
+end
+
 function MA.promote_operation(
     op::PROMOTE_IMPLEMENTED_OP,
-    F::Type{<:ScalarLike{T}},
-    G::Type{<:ScalarLike{T}},
```

Comment:
Here's another case of `ScalarLike`. This time there's only three possible cases, so we may as well just enumerate.

Reply:
#1736